bot/src/bot.ts

```diff
@@ -1,76 +1,71 @@
-import {Client} from "discord.js-light";
-import {config} from "./utils/Configuration";
-import {DiscordListener} from "./discord/DiscordListener";
-import {ReadyListener} from "./discord/listeners/ReadyListener";
-import {InteractionCreateListener} from "./discord/listeners/InteractionCreateListener";
-import {DiscordCommand} from "./discord/DiscordCommand";
-import {AirhornCommand} from "./discord/commands/AirhornCommand";
-import {AirhornMetaCommand} from "./discord/commands/AirhornMetaCommand";
-import {DiscordButton} from "./discord/DiscordButton";
-import {PlayButton} from "./discord/buttons/PlayButton";
-import {SoundboardCommand} from "./discord/commands/SoundboardCommand";
+import 'dotenv/config';
+import { PrismaClient } from '@prisma/client';
+import { ActivityType, Client, GatewayIntentBits } from 'discord.js';
+import { interactionCreateListener } from './discord/listeners/InteractionCreateListener';
+import { readyListener } from './discord/listeners/ReadyListener';
+import { Logger } from 'tslog';
+import { ILogObj } from 'tslog/dist/types/interfaces';
+import {
+  shardDisconnectListener,
+  shardReadyListener,
+  shardReconnectingListener,
+  shardResumeListener,
+} from './discord/listeners/ShardListener';
 
-export class AirhornBot {
+export const log: Logger<ILogObj> = new Logger({
+  minLevel: parseInt(process.env.MIN_LOG_LEVEL || '3', 10), // 3 is info, 2 is debug
+  hideLogPositionForProduction: true,
+});
 
-  public readonly client: Client;
-  public readonly commands: Map<string, DiscordCommand>;
-  public readonly buttons: Map<string, DiscordButton>;
+// Handle all uncaught exceptions
+process.on('uncaughtException', function (e) {
+  log.error(e);
+});
 
-  constructor() {
-    this.client = new Client({
-      ws: {
-        intents: ["GUILDS", "GUILD_VOICE_STATES"]
-      },
-      shards: "auto",
-      presence: {
-        status: "online",
-        activity: {
-          type: "PLAYING",
-          name: "airhorn.solutions"
-        }
-      },
-      cacheOverwrites: true,
-      cacheRoles: true
-    });
-    this.commands = new Map<string, DiscordCommand>();
-    this.buttons = new Map<string, DiscordButton>();
-    // Register the listeners
-    this.registerListener(new InteractionCreateListener());
-    this.registerListener(new ReadyListener());
-    // Register the commands
-    const soundKeys = Object.keys(config.sounds);
-    for (let i = 0; i < soundKeys.length; i++) {
-      this.registerCommand(new AirhornCommand(soundKeys[i]));
-    }
-    this.registerCommand(new AirhornCommand("random"));
-    this.registerCommand(new AirhornMetaCommand());
-    this.registerCommand(new SoundboardCommand());
-    // Register the buttons
-    this.registerButton(new PlayButton());
-  }
+let discordShards: number[] | 'auto' = 'auto';
```

Comment:
One of the things we're trying to do with recent bots is rely more on good ol' HTTP slash commands, and avoiding gateway if we can.  I don't expect that change here (you've already done a ton), but I want to let you know the direction we're thinking of things.  An advantage to that approach is not needing to get into the weeds of sharding, and instead being able to deploy to things like Cloud Run or Cloudflare Workers.

Reply:
See me response above on the reply about why postgres. (I include this information there.)